mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/RedeliveryPolicyTypeBuilder.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.declaration.type;
+
+import static org.mule.metadata.api.builder.BaseTypeBuilder.create;
+import static org.mule.metadata.api.model.MetadataFormat.JAVA;
+import org.mule.metadata.api.builder.BaseTypeBuilder;
+import org.mule.metadata.api.builder.ObjectTypeBuilder;
+import org.mule.metadata.api.model.MetadataType;
+
+import java.util.Map;
+
+/**
+ * Creates instances of {@link MetadataType} which represent a redelivery policy
+ *
+ * @since 1.0
+ */
+public final class RedeliveryPolicyTypeBuilder extends InfrastructureTypeBuilder {
+
+  /**
+   * @return a {@link MetadataType} representation of a redelivery policy
+   */
+  public MetadataType buildRetryPolicyType() {
+    ObjectTypeBuilder objectType = create(JAVA).objectType().id(Map.class.getName());
```

Comment:
why Map.class.getName instead of providing a more specific id like RedeliveryPolicyType ?